cmd/packer-sdc/internal/fix/gocty.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package fix
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"golang.org/x/mod/modfile"
+)
+
+const (
+	oldPath    string = "github.com/zclconf/go-cty"
+	newPath    string = "github.com/nywilken/go-cty"
+	newVersion string = "1.12.1"
+)
+
+type GoCtyFix struct {
+	OldPath, NewPath, NewVersion string
+	data                         []byte
+}
+
+// NewGoCtyFixer is an entry to the go-cty fix command
+func NewGoCtyFixer() GoCtyFix {
+	return GoCtyFix{
+		OldPath:    oldPath,
+		NewPath:    newPath,
+		NewVersion: newVersion,
+	}
+}
+
+func (f GoCtyFix) modFileFormmatedVersion() string {
+	return fmt.Sprintf("v%s", f.NewVersion)
+}
+
+func (f GoCtyFix) isUnfixed(dir string) (bool, error) {
+	modFilePath := filepath.Join(dir, "go.mod")
+	if _, err := os.Stat(modFilePath); err != nil {
+		return false, errors.Wrap(err, "failed to find go.mod file")
+	}
+	data, err := os.ReadFile(modFilePath)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to read go.mod file")
+	}
+	mf, err := modfile.Parse(modFilePath, data, nil)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse go.mod file")
+	}
+
+	// Basic go.mod with no module dependencies
+	if len(mf.Require) == 0 {
+		return false, nil
+	}
+
+	// Brute force will be better to Sort then search
+	var found bool
+	for _, req := range mf.Require {
+		if req.Mod.Path == f.OldPath {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return false, nil
+	}
+
+	if len(mf.Replace) == 0 {
+		return true, nil
+	}
+	// what happens with multiple replace
+	for _, r := range mf.Replace {
+		if r.Old.Path != f.OldPath {
+			continue
+		}
+		if r.New.Path != f.NewPath {
+			return false, errors.New("found unexpected replace for " + r.Old.Path)
+		}
+		return r.New.Version != f.modFileFormmatedVersion(), nil
+	}
+	return true, nil
+}
+
+func (f GoCtyFix) Fix(dir string, check bool) error {
+	ok, err := f.isUnfixed(dir)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return nil
+	}
+	if check {
+		fmt.Printf("%s %5s\n", "gocty", "Unfixed!")
```

Comment:
What's this `%5s`? Also since everything is static here, why the format-string? We could as well use `fmt.Println` here I'd think.

```suggestion
		fmt.Println("gocty Unfixed!")
```

Also regarding the phrasing, I think we'll confuse users by printing `unfixed`, my understanding of the `check` is to decide if we need to fix the `go.mod`, so I would suggest amending the message to clearly state that they need to run it in fix mode, something like `[packer-sdc] gocty needs to be fixed`.

Reply:
I replaced this check with a basic diff output for now. It currently looks like the following
```shell
~>  ./packer-sdc fix -diff ../../../packer/
/home/vagrant/Development/packer/go.mod:
  []uint8(
        """
        ... // 213 identical lines

        go 1.20
+
+       replace github.com/zclconf/go-cty => github.com/nywilken/go-cty v1.12.1 // add by packer-sdc fix as noted in github.com/hashicorp/packer-plugin-sdk/issues/187
+
        """
  )
```